node/src/main/scala/coop/rchain/node/web/WebApiRoutes.scala

```diff
@@ -0,0 +1,162 @@
+      //     req.handle_(webApi.lastFinalizedBlock)
+      def handle_[B](
+          f: M[B]
+      )(implicit encoder: EntityEncoder[F, B]) =
+        handleRequest[Unit, B](req, _ => f)
+    }
+
+    // Create HTML response before creating implicit JSON encoder for String
+    val htmlReponse = Ok(descriptionHtml, `Content-Type`(MediaType.text.html))
+
+    // Encoders
+    implicit val stringEncoder     = jsonEncoderOf[F, String]
+    implicit val apiStatusEncoder  = jsonEncoderOf[F, ApiStatus]
+    implicit val blockInfoEncoder  = jsonEncoderOf[F, BlockInfo]
+    implicit val lightBlockEncoder = jsonEncoderOf[F, List[LightBlockInfo]]
+    implicit val dataRespEncoder   = jsonEncoderOf[F, DataResponse]
+    implicit val prepareEncoder    = jsonEncoderOf[F, PrepareResponse]
+    // Decoders
+    implicit val signedDeployDecoder = jsonOf[F, SignedDeploy]
+    implicit val dataRequestDecoder  = jsonOf[F, DataRequest]
+    implicit val prepareDecoder      = jsonOf[F, PrepareRequest]
+
+    HttpRoutes.of[F] {
+      case GET -> Root => htmlReponse
+
+      case GET -> Root / "status" =>
+        webApi.status.handle
+
+      // Prepare deploy
+
+      case GET -> Root / "prepare-deploy" =>
+        webApi.prepareDeploy(none).handle
+
+      case req @ POST -> Root / "prepare-deploy" =>
+        req.handle[PrepareRequest, PrepareResponse](x => webApi.prepareDeploy(x.some))
+
+      // Deploy
+
+      case req @ POST -> Root / "deploy" =>
+        req.handle[SignedDeploy, String](WebApi.toSigned[M](_) >>= webApi.deploy)
+
+      // Get data
+
+      case req @ POST -> Root / "data-at-name" =>
+        req.handle[DataRequest, DataResponse](webApi.listenForDataAtName)
+
+      // Blocks
+
+      case GET -> Root / "last-finalized-block" =>
+        webApi.lastFinalizedBlock.handle
+
+      case GET -> Root / "block" / hash =>
+        webApi.getBlock(hash).handle
+
+      case GET -> Root / "blocks" =>
+        webApi.getBlocks(none).handle
+
+      case GET -> Root / "blocks" / IntVar(depth) =>
+        webApi.getBlocks(depth.some).handle
+    }
+  }
+
+  def descriptionHtml = s"""
+    <h1>RNode Web API</h1>
+    <p><b>ROOT</b>: http://localhost:40403/</p>
+    <!-- Prepare -->
+    <h2>Prepare deploy</h2>
+    <p><b><a href='api/prepare-deploy'>api/prepare-deploy</a></b></p>
+    <p><b>GET</b>: () -> Promise[PrepareResponse]</p>
+    <p><b>POST</b>: PrepareRequest -> Promise[PrepareResponse]</p>
+    <!-- Write -->
+    <h2>Deploy</h2>
+    <p><b>api/deploy</b></p>
+    <p><b>POST</b>: SignedDeploy -> Promise[String]</p>
+    <!-- Read -->
+    <h2>Get data</h2>
+    <p><b>api/data-at-name</b></p>
+    <p><b>POST</b>: DataRequest -> Promise[DataResponse]</p>
+    <!-- Blocks -->
+    <h2>Blocks</h2>
+    <p><b><a href='api/blocks'>api/blocks</a></b></p>
+    <p><b>GET</b>: () -> Promise[List[LightBlockInfo]]</p>
+    <p><b>api/blocks/(depth: Int)</b></p>
+    <p><b>GET</b>: depth -> Promise[List[LightBlockInfo]]</p>
+  """
```

Comment:
TODO: improve/update HTML, use Blaze (?), generate from trait/routes (?)

Reply:
I suggest we make restful api and leave the html things to the front end.